remill/Arch/Arch.cpp

```diff
@@ -255,6 +255,21 @@ const Arch *GetTargetArch(llvm::LLVMContext &context) {
   return arch;
 }
 
+const remill::Arch *Arch::GetModuleArch(const llvm::Module &module) {
+  auto triple = llvm::Triple(module.getTargetTriple());
+  std::string arch_name = triple.getArchName().data();
+  std::string os_name = triple.getOSName().data();
+  // Since llvm::Triple.getArchName() incorrectly returns the entire triple,
+  // just extract the first part of the triple ourselves.
+  arch_name = arch_name.substr(0, arch_name.find('-'));
+  if (arch_name.empty() || os_name.empty()) {
+    return nullptr;
```

Comment:
This returns `nullptr`, but `remill::Arch::Get` will `LOG(FATAL)` on an invalid os or arch name. I'd recommend having a more helpful yet fatal log message here where you log out the `triple.getArchName().data()` and `triple.getOSName().data()`.